src/gcinfo/gcinfoencoder.cpp

```diff
@@ -1787,34 +1842,16 @@ void GcInfoEncoder::Build()
                     *newLiveState = liveState;
                     hashMap.Set(newLiveState, (UINT32)(-1));
                 }
-
-
-                if(++callSiteIndex == m_NumCallSites)
-                    break;
-            
-                callSite = m_pCallSites[callSiteIndex];
-            }
-            else
-            {
-                UINT32 slotIndex = pCurrent->SlotId;
-                BYTE becomesLive = pCurrent->BecomesLive;
-                _ASSERTE((liveState.ReadBit(slotIndex) && !becomesLive)
-                        || (!liveState.ReadBit(slotIndex) && becomesLive));
-                liveState.WriteBit(slotIndex, becomesLive);
-                pCurrent++;
             }
         }
+        catch (GcInfoNoMemoryException& e)
+        {
+            outOfMemory = true;
+        }
 
-        // Check for call sites at offsets past the last transition
-        if (callSiteIndex < m_NumCallSites)
+        if (outOfMemory)
         {
-            UINT32 liveStateOffset = 0;
-            if (!hashMap.Lookup(&liveState, &liveStateOffset))
-            {
-                BitArray * newLiveState = new (m_pAllocator->Alloc(sizeof(BitArray))) BitArray(m_pAllocator, size_tCount);
-                *newLiveState = liveState;
-                hashMap.Set(newLiveState, (UINT32)(-1));
-            }
+            m_pNoMem();
         }
 
```

Comment:
> What are you guarding here? just the LiveStateHashTable usage?

Yes.

> What happens when they fail? Do they throw?

That depends on the allocator. In the JIT's case, I believe they call `NOMEM` and end up throwing.

> Maybe the LiveStateHashTable should do the same thing, without need for this code, or the m_pNoMem function.

It can't do so without taking a dependency on the JIT.